stdlib/in_channel.ml

```diff
@@ -63,4 +79,29 @@ let really_input_string ic len =
   | s -> Some s
   | exception End_of_file -> None
 
+let input_all ic =
```

Comment:
Possibly dumb question, but do `in_channel_length ic` and `pos_in ic` together not tell us already how many remaining bytes need to be read? Is it wasteful to have 64K buffer always, even for small files? 

I ask because I have a similar function that just does `really_input` based on the channel size, and I'm wondering if I've overlooked something.

Reply:
One issue I had in mind with using `in_channel_length` + `pos_in` is that if the channel is text-mode (on Windows), the result of `pos_in` is only approximate due to the newline conversions so it can't be used as-is. But perhaps we could use that for binary mode channels and fallback to a slower method in general (we would need to add a way to tell if a channel is binary or not --- I don't think that is possible today).

Reply:
Another issue is that `in_channel_length` returns `0` on stdin.

Reply:
> Another issue is that `in_channel_length` returns `0` on stdin.

Good point! I guess the same is true for sockets...

Reply:
I pushed a commit with this idea.